### DISCUSSION TITLE
[Request] FreshRSS

### DISCUSSION THREAD

**Discussion by dominikheinz** [tteck/Proxmox#312] [Ideas]
Hello!
I would like to request a script for FreshRSS (https://www.freshrss.org/ and https://github.com/FreshRSS/FreshRSS)
FreshRSS is a fast and lightweight open source RSS reader. The developers already provide a docker container for automated installation.
Manual installation instructions can be found here: https://freshrss.github.io/FreshRSS/en/admins/03_Installation.html

**Comment by jokerigno** [comment]
+1 on that

**Comment by Gangstrocity** [comment]
+1. I'm also looking for an RSS reader. I may end up installing this one manually.

**Reply by jokerigno** [reply]
did you find a way to install it?

**Reply by Gangstrocity** [reply]
Yeah I installed that one and CommaFeed. Ended up liking commafeed more so that's what I'm currently using.

**Reply by Gangstrocity** [reply]
https://i12bretro.github.io/tutorials/0586.html

I followed this guide for FreshRss

**Reply by jokerigno** [reply]
do you have also a guide for CommaFeed (just to check if it fits my needs).

**Reply by Gangstrocity** [reply]
Their GitHub page has a pretty easy install steps under "Download precompiled package" Just apt install default-jre before you start also If you don't have it already in your container. 

https://github.com/Athou/commafeed/#download-precompiled-package

**Reply by tteck** [reply]
## 2024-01-04

### Changed

- **CommaFeed LXC**
  - NEW Script

**Reply by Gangstrocity** [reply]
Amazing lol
Now do Vikunja. :P

**Reply by AlexGeek77** [reply]
CommaFeed LXC works great. Thanks for setting that one up.

**Reply by jokerigno** [reply]
Hi @tteck , I have some issue (seems like i'm not the only one (https://github.com/Athou/commafeed/issues/1176). Can you suggest me a way to login inside the LXC ? So i can help developer providing logs?

Thanks

**Reply by tteck** [reply]
I'm unsure what you're asking? Accessing the console is inside of the LXC.

**Reply by jokerigno** [reply]
> I'm unsure what you're asking? Accessing the console is inside of the LXC.

Sorry for being unclear. I was asking how to connect LXC using programs like WINSCP so I can use both console AND file navigation (like I already do with your Home Assistant VM ). I think it can be useful to use for example VS CODE to manage config  files instead of using console and Nano.

**Reply by raenye** [reply]
@jokerigno for winscp you need an ssh server, which is actually installed by default in the LXC.
What's not available by default is (a) root login with password; and (b) passwordless logins.

So you could either add a normal user and use it with winscp, or modify sshd config (see, e.g., [here](https://www.liquidweb.com/kb/enable-root-login-via-ssh/)).

**Comment by AlexGeek77** [comment]
```
   ______                                ______              __
  / ____/___  ____ ___  ____ ___  ____ _/ ____/__  ___  ____/ /
 / /   / __ \/ __ `__ \/ __ `__ \/ __ `/ /_  / _ \/ _ \/ __  /
/ /___/ /_/ / / / / / / / / / / / /_/ / __/ /  __/  __/ /_/ /
\____/\____/_/ /_/ /_/_/ /_/ /_/\__,_/_/    \___/\___/\__,_/

 ✗ There is currently no update path available.
```
The default update script didn't work for me so I wrote my own for the time being. It's pretty simple to upgrade. The upgrade to 4.2.0 was needed for me to use the Unread app on my iPhone to pull from the Commafeed server. 
```
#! /bin/bash
apt update
apt list --upgradeable
apt upgrade
cd /opt/commafeed
echo "Stopping commafeed service"
systemctl stop commafeed
echo "Backing up JAR file"
mv commafeed.jar commafeed.jar.old
wget -nv --show-progress https://github.com/Athou/commafeed/releases/latest/download/commafeed.jar
echo "Restarting commafeed service"
systemctl start commafeed
echo "Upgrade complete"
```

**Reply by tteck** [reply]
The `mv commafeed.jar commafeed.jar.old` line will result in an error if the script is run a second time without deleting or moving the existing `commafeed.jar.old` file. The `mv` command will not overwrite an existing file.

```
echo "Backing up JAR file"
if [[ -f commafeed.jar.old ]]; then
    rm commafeed.jar.old
fi
mv commafeed.jar commafeed.jar.old
```

**Reply by tteck** [reply]
Update path created https://github.com/tteck/Proxmox/commit/edd6eacdd3763c7eac7df9941505596851d4cd6e

**Comment by jsspen** [comment]
+1 for a FreshRSS script. I'm going to try out CommaFeed in the meantime but I've been running a FreshRSS docker for a while now on my old server and would prefer to just migrate to an LXC on my new server instead of switching to a whole new reader.

**Comment by spazbg** [comment]
I also really need FreshRSS
